### PR TITLE
Onboarding

### PR DESCRIPTION
**Pull request by sagar5534**
Story details: https://app.clubhouse.io/chrono/story/474

Couple of things about this PR:

- I'm saving a boolean locally on users devices to decide if the onboarding page should be shown or not (New Users)
- Currently, only the Lottie Animations are working. I may have to go back and improve the Lottie Animation Controller (It's not really worth the time diving into it just yet. I just made a simple wrapper for the time being) 
- Gifs are not supported yet. 
- The images at the moment are placeholders until we can take IOS specific screenshots

@harigovind1998 To test locally multiple times, Go into ViewRouter and uncomment line 19! Big help

https://user-images.githubusercontent.com/15317925/123210796-85cb3b80-d490-11eb-97f2-8bbb3565664b.MP4

### CODE REVIEW

File: Chrono/Chrono/Assets.xcassets/Onboarding/onboard_circuits.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "filename" : "ezgif.com-gif-maker (1).png",
```
Comment (harigovind1998) [inline Chrono/Chrono/Assets.xcassets/Onboarding/onboard_circuits.imageset/Contents.json:4]: Change the name of the file to something more descriptive.

File: Chrono/Chrono/Assets.xcassets/Onboarding/onboard_dark.imageset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "filename" : "ezgif.com-gif-maker.png",
```
Comment (harigovind1998) [inline Chrono/Chrono/Assets.xcassets/Onboarding/onboard_dark.imageset/Contents.json:4]: Here as well

File: Chrono/Chrono/Assets.xcassets/SplashScreen/SplashScreen_LightGrad1.colorset/Contents.json [hunk 3]
```diff
@@ -5,9 +5,9 @@
         "color-space" : "srgb",
         "components" : {
           "alpha" : "1.000",
-          "blue" : "0xFF",
-          "green" : "0x51",
-          "red" : "0x7E"
+          "blue" : "1.000",
```
Comment (harigovind1998) [inline Chrono/Chrono/Assets.xcassets/SplashScreen/SplashScreen_LightGrad1.colorset/Contents.json:8]: Are scaled values preferred over Hexcodes?

File: Chrono/Chrono/ChronoApp.swift [hunk 4]
```diff
@@ -5,20 +5,18 @@
 //  Created by Sagar on 2021-06-15.
 //
 
-import SwiftUI
 import Firebase
+import SwiftUI
```
Comment (harigovind1998) [inline Chrono/Chrono/ChronoApp.swift:9]: Is there an autoformatted you're using? How do I use it?

File: Chrono/Chrono/Common Elements/LottieView.swift [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+struct LottieView: UIViewRepresentable {
+    typealias UIViewType = UIView
+    var filename: String
+
+    func makeUIView(context _: UIViewRepresentableContext<LottieView>) -> UIView {
+        let view = UIView(frame: .zero)
+
+        let animationView = AnimationView()
+        let animation = Animation.named(filename)
```
Comment (harigovind1998) [inline Chrono/Chrono/Common Elements/LottieView.swift:19]: I think we should add a default animation if the provided filename is incorrect, or add a null check to ensure that if the file isn't found, we don't crash.

File: Chrono/Chrono/ViewRouter.swift [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import SwiftUI
+
+enum ViewType {
+    case main, onboarding
+}
+
+class ViewRouter: ObservableObject {
+    init() {
+        // Use this only to test!!
+        UserDefaults.standard.set(false, forKey: "didLaunchBefore")
```
Comment (harigovind1998) [inline Chrono/Chrono/ViewRouter.swift]: Should this be commented out? Im guessing right now didLaunchBefore gets set to false whenever the app opens.

File: Chrono/Chrono/ViewRouter.swift [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+            currentPage = .onboarding
+        } else {
+            currentPage = .main
+        }
+    }
+
+    @Published var currentPage: ViewType
+}
+
+struct MotherView: View {
```
Comment (harigovind1998) [inline Chrono/Chrono/ViewRouter.swift:30]: I like this name XD

**Review by harigovind1998 — CHANGES_REQUESTED** [review]

File: Chrono/Chrono/ViewRouter.swift (hunk 6 above)
Comment (sagar5534) [reply Chrono/Chrono/ViewRouter.swift]: Fixed

File: Chrono/Chrono/ChronoApp.swift (hunk 4 above)
Comment (sagar5534) [reply Chrono/Chrono/ChronoApp.swift:9]: Ill show you!

File: Chrono/Chrono/Common Elements/LottieView.swift (hunk 5 above)
Comment (sagar5534) [reply Chrono/Chrono/Common Elements/LottieView.swift:19]: I'm actually thinking of revisiting this wrapper. Maybe look for a better library or something! So I'll note this stuff down and we can try to implement it into the better Lottie impl.

File: Chrono/Chrono/Assets.xcassets/SplashScreen/SplashScreen_LightGrad1.colorset/Contents.json (hunk 3 above)
Comment (sagar5534) [reply Chrono/Chrono/Assets.xcassets/SplashScreen/SplashScreen_LightGrad1.colorset/Contents.json:8]: I don't think there's much of a preferred method. Swift will convert any passed value into a UIColor

File: Chrono/Chrono/Assets.xcassets/Onboarding/onboard_circuits.imageset/Contents.json (hunk 1 above)
Comment (sagar5534) [reply Chrono/Chrono/Assets.xcassets/Onboarding/onboard_circuits.imageset/Contents.json:4]: Isnt an issue(Apple does a weird thing) Ill show you on call

**Review by harigovind1998 — APPROVED** [review]

**Pull request merged**